compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -129,11 +135,25 @@ public void visit(BLangSimpleVarRef varRef) {
         }
 
         if (!this.unresolvedConstants.containsKey(constSymbol)) {
+            if (cannotResolveConstants.contains(constSymbol)) {
+                this.result = null;
+                return;
+            }
+            cannotResolveConstants.add(constSymbol);
```

Comment:
```suggestion
            this.cannotResolveConstants.add(constSymbol);
```

Let's update wherever applicable, helps with readability.